myia/api.py

```diff
@@ -25,10 +25,16 @@ class MyiaFunction:
 
     """
 
-    def __init__(self, fn, specialize_values=[]):
+    def __init__(self, fn, specialize_values=[],
+                 backend=None, backend_options=None):
         """Initialize a MyiaFunction."""
         self.fn = fn
         self.specialize_values = set(specialize_values)
+        pp = standard_pipeline.configure({
+            'compile.backend': backend,
+            'compile.backend_options': backend_options
+        })
+        self.pip = pp.make()
```

Comment:
It may cause issues if `make` is called here, because that means all recompiles of the function will reuse the same manager, the same inferrer, etc. and I'm not confident that'll always work.